### PR TITLE
Unobfuscate execution plans for Postgres

### PR DESCRIPTION
**Pull request by alexbarksdale**
### What does this PR do?

Reveals useful information within a SQL execution plan for Postgres.

### Motivation

In the Database Monitoring product, we obfuscate data within the SQL execution plans to prevent revealing important information in customer's data. However, we found that we're overly obfuscating some useful information in execution plans for Postgres.

### Additional Notes

We have to make sure we do not reveal important information in customer's data, so we must proceed with caution when we unobfuscate SQL execution plans. 

### Describe how to test your changes

1. Added more tests for Postgres SQL execution plans that contains the newly added fields. You can run them via: `invoke test --targets=./pkg/trace/obfuscate --skip-linters`
2. These changes will sit in staging for a few days to ensure important data isn't revealed.

### Checklist

- [x] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.

### CODE REVIEW

File: pkg/collector/python/datadog_agent.go
Comment (justiniso) [inline pkg/collector/python/datadog_agent.go:295]: I see some fields that should not be fully excluded, but should pass through an obfuscation layer (i.e. `ObfuscateSQLValues`):

* `Filter`
* `Index Cond`

**Review by dujuku — CHANGES_REQUESTED** [review]
Discussed offline. There are some additional fields we want to add before merging this.

File: pkg/collector/python/datadog_agent.go
Comment (alexbarksdale) [reply pkg/collector/python/datadog_agent.go:297]: New fields added: d041839969802f77ddeaae96e51bbbd4fe06acb5

File: pkg/collector/python/datadog_agent.go
Comment (alexbarksdale) [reply pkg/collector/python/datadog_agent.go]: Updated: 390f87df15a928b5ed60a22c3de4afb6930623a8

**Review by sgnn7 — APPROVED** [review]

File: pkg/collector/python/datadog_agent.go
Comment (justiniso) [inline pkg/collector/python/datadog_agent.go:267]: These are also supposed to be in alphabetical order 😉

File: releasenotes/notes/add-more-postgres-exec-plans-adcfd187630acd5d.yaml
Comment (justiniso) [inline releasenotes/notes/add-more-postgres-exec-plans-adcfd187630acd5d.yaml:11]: nit: let's remove "safe"

I think it gives the wrong impression as anything we choose to expose is implicitly deemed safe.

File: pkg/trace/obfuscate/testdata/json_tests.xml
Comment (justiniso) [inline pkg/trace/obfuscate/testdata/json_tests.xml:900]: Looks like this one can also be unobfuscated.

**Review by dujuku — COMMENTED** [review]
Thanks for the updates @alexbarksdale 👍 

Sorry to be a bit annoying, but I'm going challenge you to go further with this one and try to make the field list as exhaustive as possible. We can always update the agent later, but it's much better if we can make the updates now while it's on our minds and never have to think about it again.

I encourage you to look at the Postgres source: https://github.com/postgres/postgres/blob/master/src/backend/commands/explain.c

From looking at the code there, I can see quite a few fields which can be safely exposed defined by the classes:

* `ExplainPropertyFloat`
* `ExplainPropertyBool`
* `ExplainPropertyInteger`

Some might require more judgment like:

* `ExplainPropertyList`
* `ExplainPropertyText`
* etc.

When in doubt, feel free to ask for a second pair of eyes and/or add it to the `ObfuscateSQLValues` slice. There may also be some test cases which you can copy over.

File: pkg/collector/python/datadog_agent.go
Comment (justiniso) [inline pkg/collector/python/datadog_agent.go:325]: I'm not entirely sure but should this one pass through the obfuscator?

https://github.com/postgres/postgres/blob/master/src/backend/commands/explain.c#L2400-L2428

File: pkg/collector/python/datadog_agent.go
Comment (justiniso) [inline pkg/collector/python/datadog_agent.go:330]: Do you have an example of these?

File: pkg/trace/obfuscate/testdata/json_tests.xml
Comment (justiniso) [inline pkg/trace/obfuscate/testdata/json_tests.xml:792]: I haven't looked through all of these, but can you try to find some realistic examples for the tests? The type on this one is actually incorrect: https://github.com/postgres/postgres/blob/master/src/backend/commands/explain.c#L2444-L2497

File: pkg/collector/python/datadog_agent.go
Comment (alexbarksdale) [reply pkg/collector/python/datadog_agent.go:330]: Discussed offline, but for future readers. `Grouping Sets` looks to only reveal column names, so obfuscation is unnecessary.
Example:
```
SELECT
    c1,
    c2,
    aggregate_function(c3)
FROM
    table_name
GROUP BY
    GROUPING SETS (
        (c1, c2),
        (c1),
        (c2),
        ()
);
```

File: pkg/trace/obfuscate/testdata/json_tests.xml
Comment (alexbarksdale) [reply pkg/trace/obfuscate/testdata/json_tests.xml]: Improved: https://github.com/DataDog/datadog-agent/pull/8411/commits/7c77d58e9dff3c11dfb740bd7e8e8fc1e709115d

File: pkg/trace/obfuscate/testdata/json_tests.xml
Comment (justiniso) [inline pkg/trace/obfuscate/testdata/json_tests.xml:931]: nit: some spacing issues

**Review by dujuku — APPROVED** [review]
Thanks for all of the updates, this LGTM now 👍

File: pkg/collector/python/datadog_agent.go
Comment (alexbarksdale) [reply pkg/collector/python/datadog_agent.go]: Discussed offline, but a note for future readers. We were unable to discover what values these two fields provide, so we've removed them for this iteration.

**Review by gbbr — APPROVED** [review]
Approving strictly for `./pkg/trace/...`

File: pkg/trace/obfuscate/testdata/json_tests.xml
Comment (djova) [inline pkg/trace/obfuscate/testdata/json_tests.xml:330]: Should undo unrelated formatting changes

**Review by djova — APPROVED** [review]

**Pull request merged**
